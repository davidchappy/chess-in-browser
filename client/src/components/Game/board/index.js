import React from 'react';
import './board.css';
import Tile from './tile';

const Board = (props) => {

  // create 2-dim. array with tile rows from game.board data
  const tileNames = Object.keys(props.board);
  let row = [], allRows = [];
  for(let i=0; i<tileNames.length; i++) {
    let tile = tileNames[i];
    if(i === 0 || i % 8 !== 0) {
      row.push(tile);
    } else {
      allRows.push(row);
      row = [];
      row.push(tile);
    }
    if (i === tileNames.length-1) {
      allRows.push(row);
    } 
  }

  const allPieces = props.pieces;
  // map through 2dim array of tiles and return rows of Tiles
  const tiles = allRows.map((rowArray, i) => {
    let rowType = i % 2 === 0 ? 0 : 1;
    let row = rowArray.map((tile, i) => {
      const color = i % 2 === rowType ? 'light' : 'dark';
      let piece = null;
      if(props.board[tile] !== "") {
        for(let pieceNo in allPieces) {
          if(allPieces[pieceNo].position === tile) {
            piece = allPieces[pieceNo];
          }
        }
      }
      const selectable = props.selectableTiles.includes(tile) ? true : false; 
      return (<Tile {...props} tile={tile} key={i} 
                    color={color} piece={piece} 
                    selectable={selectable} />)
    })
    return (
      <div key={i} className="tileRow">
        {row}
      </div>
    )
  });

  return (
    <div className="Board">
      <div className="tiles">
        {tiles}
      </div>
    </div>
  )

}

export default Board;