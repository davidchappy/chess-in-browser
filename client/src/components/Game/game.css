.Game {
  background-color: #D19255;
  height: 100vh;
  width: 100vw;
  text-align: center;
}