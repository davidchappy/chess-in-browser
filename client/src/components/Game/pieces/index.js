import React from 'react';
import Piece from './piece';

const Pieces = (props) => {
  return (
    <div>
      <Piece type="king"/>
    </div>
  )
};

export default Pieces;