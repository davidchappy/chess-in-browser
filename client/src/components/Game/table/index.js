import React from 'react';
import Board from '../board';
import './table.css';

const Table = (props) => {
  return (
    <div className="Table">
      <Board />
    </div>
  )
};

export default Table;