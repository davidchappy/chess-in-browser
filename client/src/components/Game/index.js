import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './Game.css';
import Table from './table/';
import Header from '../shared/header';
import { selectPiece, movePiece } from '../../actions/chessActions';

export class Game extends Component {
  constructor(props) {
    super(props);
  
    this.state = {};
  }
  
  render() {
    return (
      <div className="Game">
        <Header {...this.props} headerSize="small" />
        <Table {...this.props} />
      </div>  
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    game: state.chess.game,
    white: state.chess.white,
    black: state.chess.black,
    moving: state.chess.moving,
    selectedPiece: state.chess.selectedPiece,
    selectableTiles: state.chess.selectableTiles,
    fetching: state.chess.fetching,
    fetched: state.chess.fetched
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({
  selectPiece,
  movePiece
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Game)
