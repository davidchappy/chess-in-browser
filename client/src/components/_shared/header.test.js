import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Header from './header';
import Game from './header';

const props = {
  white: {
    is_playing: true
  },
  black: {
    is_playing: false
  },
  game: {
    status: 'check'
  }
}

describe('rendering Header (Snapshot Test)', () => {
  const component = renderer.create(<Header {...props} />);
  const json = component.toJSON();

  it('renders without crashing', () => {
    expect(json).toMatchSnapshot();
  });
});

describe('rendering Header in Welcome view', () => {
  const welcomeHeaderWrapper = mount(
    <Header welcome={true}/>
  );

  it('renders welcome text', () => {
    const h2 = welcomeHeaderWrapper.find('h2');
    expect(h2.text()).toBe('Welcome to Chess');
  });

  it('renders a large logo', () => {
    const logo = welcomeHeaderWrapper.find('img.chess-logo.large');
    expect(logo.length).toBe(1);
  });
});


describe('rendering Header in Game view', () => {
  const gameHeaderWrapper = mount(
    <Header headerSize='small' {...props} />
  );

  it('renders not heading text', () => {
    const h2 = gameHeaderWrapper.find('h2');
    expect(h2.length).toBe(0);
  });


  it('renders a logo', () => {
    const logo = gameHeaderWrapper.find('img.chess-logo.small');
    expect(logo.length).toBe(1);
  });
});