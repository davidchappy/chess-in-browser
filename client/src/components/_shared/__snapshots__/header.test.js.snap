// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Header (Snapshot Test) renders without crashing 1`] = `
<div
  className="chess-header large"
>
  <img
    alt="logo"
    className="chess-logo loaded large"
    src="logo.svg"
  />
  <div
    className="Messages"
  >
    <p>
      <span>
        Now Playing: 
      </span>
      <span>
          |  Check!
      </span>
    </p>
  </div>
</div>
`;
